webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/HealthcheckInfo.scala

```diff
@@ -0,0 +1,133 @@
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.web.snippet
+
+import bootstrap.liftweb.RudderConfig
+import com.normation.rudder.services.healthcheck.HealthcheckResult
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Critical
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Ok
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Warning
+import com.normation.rudder.services.healthcheck.HealthcheckUtils.compareCheck
+import com.normation.zio._
+import net.liftweb.common.Loggable
+import net.liftweb.http.DispatchSnippet
+
+import scala.xml.NodeSeq
+
+class HealthcheckInfo(
+) extends DispatchSnippet with Loggable{
+
+  def dispatch = {
+    case "render"     => healthcheckInfo
+    case "renderIcon" => healthcheckInfoIcon
+  }
+
+  def isCriticalOrWarning(checks: List[HealthcheckResult]): (Boolean, Boolean) = {
+    val isCritical: Boolean = checks.exists(
+      x => x match {
+        case Critical(_) => true
+        case _           => false
+      }
+    )
+      val isWarning: Boolean = checks.exists(
+        x => x match {
+          case Warning(_) => true
+          case _          => false
+        }
+      )
+    (isCritical, isWarning)
+  }
+
+  def healthcheckInfo(html: NodeSeq) ={
+    def notifHtml (notifClass:String , notifTitle:String, checks: List[HealthcheckResult]): NodeSeq ={
+      val liChecks = checks.map(c => {
+        val circle = c match {
+          case _:Warning  => "<div class='circle-notif warning-light-notif'></span>"
+          case _:Critical => "<div class='circle-notif critical-light-notif'></span>"
+          case _          => ""
+        }
+        "<li class='notif-msg-hc'><span>" + c.msg + circle + "</span></li>"
+      })
+      val checksMenu = "<ul class='menu'>" + liChecks.mkString("\n") + "</ul>"
+
+      val tooltipContent = "<h4><i class='fa fa-warning'></i> "+ notifTitle +"</h4><div>" + checksMenu + "</div>"
+      <li
+      class={"plugin-warning " + notifClass}
+      data-toggle="tooltip"
+      data-placement="bottom"
+      data-html="true"
+      data-original-title={tooltipContent}
+      >
+        <a href="/secure/utilities/healthcheck"><span class="fa fa-heart"></span></a>
+      </li>
+    }
+
+    (for {
+      checks <- RudderConfig.healthcheckNotificationService.healthcheckCache.get
+    } yield {
+      val c = checks.filter(_ match {
+        case _:Ok => false
+        case _    => true
+      }).sortWith((c1, c2) => compareCheck(c1, c2))
+
+      isCriticalOrWarning(c) match {
+        case (true, _)      => notifHtml("critical", "There is an anomaly that required your attention", c)
+        case (false, true)  => notifHtml("warning", "Something may cause an anomaly", c)
+        case (false, false) => NodeSeq.Empty
+      }
+    }).runNow
```

Comment:
OK, you don't even want to look for the criticity & all. You just want to sort checks (critical>warning>info), and filter-collect-transform result (only warning and critical):
 def compare(c1: CheckResult, c2: CheckResult): Int = (c1, c2) match { .... }
val sorted = checks.sortWith(compare)
sorted match {
case Nil => None
case h::_ => //here, head is the max level, we get cssClass and transform
  val transformed = sorted.flatMap { 
   case Warning(msg, ..) => Some(<li class='notif-msg-hc'><span>{ c.msg }<div class='circle-notif warning-light-notif'></span></span></li>)
   case Critical(msg, ...) => ....
case _ => None
}
 //now you have cssClass and transformed value, just put everything together
 ... 
 // Return a Some(all piece neede)
}
```
And here, if you hava a  Some, then there is notif to display, and you already have the content, else NodeSeq.empty